tools/server/server.cpp

```diff
@@ -3141,12 +3110,12 @@ struct server_context {
 
                                     SLT_DBG(slot, "trying to reuse chunks with size > %d, slot.n_past = %d\n", params_base.n_cache_reuse, slot.n_past);
 
-                                    while (head_c < slot.cache_tokens.size() &&
-                                           head_p < prompt_tokens.size()) {
+                                    while (head_c < (size_t)slot.n_cache_tokens() &&
+                                           head_p < prompt_tokens.n_tokens()) {
 
                                         size_t n_match = 0;
-                                        while (head_c + n_match < slot.cache_tokens.size() &&
-                                               head_p + n_match < prompt_tokens.size()     &&
+                                        while (head_c + n_match < (size_t)slot.n_cache_tokens() &&
+                                               head_p + n_match < prompt_tokens.n_tokens() &&
```

Comment:
This logic no longer seems correct. At least cache `--cache-reuse 256` no longer works. I think the `slot.n_past` is no longer used correctly to initialize the `head_c` / `head_p`.

Reply:
Hmm, or maybe we can disable ctx shift if chunk reuse is enabled, so we no longer need to worry about `slot.n_past` here?

Another approach is to move this whole logic into `server_tokens`. But tbh I haven't yet fully understand this context chunk logic, so I'm not sure what is the best way. Feel free to suggest a fix.